rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/Costs.scala

```diff
@@ -53,6 +53,10 @@ trait Costs {
   def toByteArrayCost[T <: GeneratedMessage](a: T): Cost = Cost(a.serializedSize)
   //TODO: adjust the cost of size method
   def sizeMethodCost(size: Int): Cost = Cost(size)
+  def sliceCost(from: Int, to: Int): Cost = {
+    val diff = to - from
+    if (diff <= 0) Cost(0) else Cost(diff)
+  }
```

Comment:
I think the cost of slicing should be `from + (to - from) == to`, becuase we probably first need to pass `from` elements to then read the `to - from` of the ones we're interested in.

Also, we should probably not charge 100 for [1,2,3].slice(2, 100) (the cost should be parameterized - and capped by - collection size).

Did we do any range checking for costs so far? If not, I'd add it here and file a JIRA for adding it in other places.

Reply:
> Also, we should probably not charge 100 for [1,2,3].slice(2, 100) (the cost should be parameterized - and capped by - collection size).

good call.

> Did we do any range checking for costs so far? If not, I'd add it here and file a JIRA for adding it in other places.

I am writing tests at the very moment.